cmd/kubeadm/app/phases/upgrade/preflight.go

```diff
@@ -126,3 +128,45 @@ func checkMigration(client clientset.Interface) error {
 	}
 	return nil
 }
+
+// EnforceRequirements verifies that it's okay to upgrade and then returns the variables needed for the rest of the procedure
+func EnforceRequirements(client clientset.Interface, cfg *kubeadmapi.InitConfiguration, ignorePreflightErrorsSet sets.String, featureGatesString string, dryRun bool) error {
+	// Check if the cluster is self-hosted
+	if IsControlPlaneSelfHosted(client) {
+		return errors.New("cannot upgrade a self-hosted control plane")
+	}
+
+	// Fetch the configuration from a file or ConfigMap and validate it
+	fmt.Println("[upgrade/config] Making sure the configuration is correct:")
+
+	var err error
+	// If features gates are passed to the command line, use it (otherwise use featureGates from configuration)
+	if featureGatesString != "" {
+		cfg.FeatureGates, err = features.NewFeatureGate(&features.InitFeatureGates, featureGatesString)
+		if err != nil {
+			return errors.Wrap(err, "[upgrade/config] FATAL")
+		}
+	}
+
+	// Check if feature gate flags used in the cluster are consistent with the set of features currently supported by kubeadm
+	if msg := features.CheckDeprecatedFlags(&features.InitFeatureGates, cfg.FeatureGates); len(msg) > 0 {
+		for _, m := range msg {
+			fmt.Printf("[upgrade/config] %s\n", m)
+		}
+		return errors.New("[upgrade/config] FATAL. Unable to upgrade a cluster using deprecated feature-gate flags. Please see the release notes")
+	}
+
+	// Ensure the user is root
+	klog.V(1).Info("running preflight checks")
+	if err := RunPreflightChecks(client, ignorePreflightErrorsSet, &cfg.ClusterConfiguration); err != nil {
+		return err
+	}
+
+	// Run healthchecks against the cluster
+	if err := CheckClusterHealth(client, &cfg.ClusterConfiguration, ignorePreflightErrorsSet); err != nil {
+		return errors.Wrap(err, "[upgrade/health] FATAL")
```

Comment:
the function name is still a bit odd - EnforceRequirements().
it is a function that performs a collection of checks.

`RunClusterChecks()` seems better to me.

Reply:
Ok. yeah. I mean. it was a function that did EVERYTHING before. I did not change its name though I ripped half of it's functionality out.
I like `RunClusterChecks` though. will see if I don't think of anything better, if not I will change it too that.